webapp/src/components/modals/actionModal.tsx

```diff
@@ -0,0 +1,173 @@
+    setVisibility: React.Dispatch<React.SetStateAction<boolean>>;
+    config: Configs[];
+    configIndex: number;
+}
+
+const ActionModal = ({visibility, setVisibility, config, configIndex}: Props) => {
+    const [show, setShow] = useState(false);
+
+    const [attachmentMessageAvailable, setAttachmentMessageAvailable] = useState(false);
+
+    const actionsLength = config[configIndex]?.actions?.length ?? 0;
+
+    const checkAttachmentMessage = () => {
+        if (config[configIndex]?.attachmentMessage) {
+            if (config[configIndex]?.attachmentMessage?.length === 0) {
+                setAttachmentMessageAvailable(false);
+            } else {
+                setAttachmentMessageAvailable(config[configIndex]?.attachmentMessage?.[0] !== '');
+            }
+        }
+    };
+
+    useEffect(() => {
+        checkAttachmentMessage();
+        setShow(visibility);
+    }, [visibility]);
+
+    const handleClose = () => {
+        setShow(false);
+        setVisibility(false);
+    };
+    return (
+        <div>
+            <Modal
+                className='customModal'
+                show={show}
+                onHide={handleClose}
+            >
+                <Modal.Header closeButton={false}>
+                    <Modal.Title>{'Actions'}</Modal.Title>
+                </Modal.Header>
+
+                <Modal.Body className='customModalBody'>
+                    {attachmentMessageAvailable || (config[configIndex]?.actions && actionsLength > 0) ? (<>
+                        {attachmentMessageAvailable ? (
+                            <Form>
+                                <Form.Group className='form-group'>
+                                    <Form.Label>{'Attachment Message'}</Form.Label>
+                                    <Form.Control
+                                        type='long-text'
+                                        value={config[configIndex].attachmentMessage ?? ''}
+                                        placeholder=''
+                                        aria-label='Disabled input example'
+                                        readOnly={true}
+                                    />
+                                </Form.Group>
+                            </Form>
+                        ) : (<p>{'No attachment message configured'}</p>)
+                        }
+                        {config[configIndex]?.actions && actionsLength > 0 ? (
+                            <div>
+                                <Form>
+                                    <Form.Group className='action-group'>
+                                        <Form.Label>{'Actions'}</Form.Label>
+                                    </Form.Group>
+                                </Form>
+                                <div className='listTable'>
+                                    <Table
+                                        striped={true}
+                                        className='listTable'
```

Comment:
Parent and child have the same className. Is this what you want to do?

Reply:
yes